core/chain_manager.go

```diff
@@ -92,6 +92,7 @@ type ChainManager struct {
 	futureBlocks *BlockCache
 
 	quit chan struct{}
+	wg   sync.WaitGroup
```

Comment:
I made these changes while working out a crash in the tests together with Gustav. The blockpool was crashing because it didn't expect to be started and stopped in short succession.

While looking at the source of the events, I found that `ChainManager` now has a background goroutine which is never stopped. The changes ensure that it is stopped, and also that stopping blocks until the loop is down.

> Second I suggest you find some other way to solve this.

`sync.WaitGroup` is the canonical way to wait for one or more goroutines to end. But it could also be done with a channel instead, if you prefer that.